src/components/SnapshotDetailView.jsx

```diff
@@ -89,8 +97,18 @@ export class SnapshotDetailView extends React.PureComponent {
     dispatch(removeCustodianFromSnapshot(snapshot.id, removeableEmail));
   };
 
+  handleFilterDatasets = (limit, offset, sort, sortDirection, searchString) => {
+    const { snapshot } = this.props;
+    let datasets = snapshot.source.map((s) => s.dataset);
+    datasets = datasets.filter((d) => d.name.toLowerCase().includes(searchString.toLowerCase()));
+    datasets = _.orderBy(datasets, sort, sortDirection);
+    datasets = _.take(_.drop(datasets, offset), limit);
```

Comment:
would this be more readable as a chain rather than a reassignment?

Reply:
It might be most readable to separate each method out on a new line, but instead of reassigning, give a more descriptive label to each array: 
```
const filtered = datasets.filter((d) => d.name.toLowerCase().includes(searchString.toLowerCase()));
const sorted = _.orderBy(filtered, sort, sortDirection);
const paged = _.take(_.drop(sorted, offset), limit);
```